src/sidebar/util/annotation-metadata.js

```diff
@@ -206,8 +207,10 @@ export function isHighlight(annotation) {
   if (annotation.$highlight) {
     return true;
   }
+  // treat like @type Annotation from here
+  const _annotation = /**@type {Annotation}*/ (annotation);
```

Comment:
This cast hides a possible error if the parameter is an `AnnotationData` rather than an `Annotation`. If any of the code below tried to access a property of the object that only existed on an `annotation`, that could have unexpected behaviour.

One way to resolve this problem would be to modify the code in `src/sidebar/services/annotations.js` so that it only calls `isHighlight` with an `Annotation`:

```diff
diff --git a/src/sidebar/services/annotations.js b/src/sidebar/services/annotations.js
index ef7f07d7d..e55dc96f9 100644
--- a/src/sidebar/services/annotations.js
+++ b/src/sidebar/services/annotations.js
@@ -60,18 +60,12 @@ export default function annotationsService(api, store) {
     // as it has not been persisted to the service.
     const $tag = generateHexString(8);
 
-    let permissions = defaultPermissions(userid, groupid, defaultPrivacy);
-
-    // Highlights are peculiar in that they always have private permissions
-    if (metadata.isHighlight(annotationData)) {
-      permissions = privatePermissions(userid);
-    }
-
-    return Object.assign(
+    /** @type {Annotation} */
+    const annotation = Object.assign(
       {
         created: now.toISOString(),
         group: groupid,
-        permissions,
+        permissions: defaultPermissions(userid, groupid, defaultPrivacy),
         tags: [],
         text: '',
         updated: now.toISOString(),
@@ -79,8 +73,15 @@ export default function annotationsService(api, store) {
         user_info: userInfo,
         $tag: $tag,
       },
-      /**@type {Object}*/ (annotationData)
+      annotationData
     );
+
+    // Highlights are peculiar in that they always have private permissions
+    if (metadata.isHighlight(annotation)) {
+      annotation.permissions = privatePermissions(userid);
+    }
+
+    return annotation;
   }
 
   /**
```

With this diff, no changes to `isHighlight` are needed. Note that in that diff I have assumed that the issue mentioned above with the cast from `AnnotationData` to `Annotation` has been fixed.

Reply:
👍

Reply:
There are still unnecessary changes here.